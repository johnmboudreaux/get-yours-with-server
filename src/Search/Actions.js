import cleanAmazonData from '../Utils/CleanAmazonData';

export const loadProductSuccess = (products) => ({
  type: 'LOAD_PRODUCTS',
  products
});

export const loadProducts = (keyword) => {
  return (dispatch) => {
    return fetch(`/api?category=TopSellers&keyword=TopSellers ${keyword}`)
      .then(results => {
        return results.json();
      })
      .then(productData => {
        let cleanData = productData.length ?
          productData.map(product => cleanAmazonData(product)) : [];
        dispatch(loadProductSuccess(cleanData));
      });
  };
};
