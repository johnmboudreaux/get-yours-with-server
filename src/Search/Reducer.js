// const Reducer = (state = [], action) => {
//   switch (action.type) {
//   case 'LOAD_BABY_PRODUCTS':
//     return action.babyProducts;
//   default:
//     return state;
//   }
// };
//
// export default Reducer;
