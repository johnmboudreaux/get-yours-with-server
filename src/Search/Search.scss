.header-div {
  display: flex;
  flex-wrap: nowrap;
  height: 40px;
}

.header-input {
  font-size: .90em;
  height: 40px;
  width: 69%;
}

.header-button {
  background-color: #02a9ea;
  font: bold 11px Arial;
  font-size: .9em;;
  height: 45px;
  width: 30%;
  margin-left: 1px;
}

.header-button:focus,
.header-button:hover {
  background-color: #171b1e;
  color: #02a9ea;
}
