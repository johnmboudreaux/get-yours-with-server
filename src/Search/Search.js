import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as Actions from './Actions';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';

export class Search extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      inputValue: ""
    };
    this.updateInputValue = this.updateInputValue.bind(this);
    this.searchClick = this.searchClick.bind(this);
  }
  updateInputValue(event) {
    this.setState({inputValue: event.target.value});
  }

  searchClick() {
    this.props.actions.loadProducts(this.state.inputValue);
  }

  render() {
    return (
      <header className="header-div">
        <input
          className="header-input"
          type={'text'}
          placeholder="Search For Products Here"
          onChange={this.updateInputValue}/>
        <button
          className="header-button"
          onClick={this.searchClick}>
          Search Baby Products
        </button>
      </header>
    );
  }

}

Search.propTypes = {
  babyProducts: PropTypes.array,
  actions: PropTypes.object
};

const mapStateToProps = (state, ownProps) => {
  return {
    products: state.products
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators(Actions, dispatch)
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Search);
