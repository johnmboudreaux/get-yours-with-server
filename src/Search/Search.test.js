import { shallow } from 'enzyme';
import { Search } from './Search';
import React from 'react';

describe('Search snapshot', () => {

  it('should always match the snapshot', () => {
    const wrapper = shallow(<Search />);

    expect(wrapper).toMatchSnapshot();
  });
});
