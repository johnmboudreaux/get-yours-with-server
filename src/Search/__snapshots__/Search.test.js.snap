// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="header-input"
          onChange={undefined}
          placeholder="Search For Products Here"
          type="text"
          value={undefined}
        />,
        <button
          className="header-button"
          onClick={undefined}
        >
          Search Products
        </button>,
      ],
      "className": "header-div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "header-input",
          "onChange": undefined,
          "placeholder": "Search For Products Here",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Search Products",
          "className": "header-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Search Products",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="header-input"
            onChange={undefined}
            placeholder="Search For Products Here"
            type="text"
            value={undefined}
          />,
          <button
            className="header-button"
            onClick={undefined}
          >
            Search Products
          </button>,
        ],
        "className": "header-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "header-input",
            "onChange": undefined,
            "placeholder": "Search For Products Here",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Search Products",
            "className": "header-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Search Products",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
